subprojects/internal-testing/src/main/groovy/org/gradle/util/TestPrecondition.groovy

```diff
@@ -73,6 +73,10 @@ enum TestPrecondition implements org.gradle.internal.Factory<Boolean> {
     LINUX({
         OperatingSystem.current().linux
     }),
+    HAS_DOCKER({
+        // TODO: should use a better detection strategy
```

Comment:
FYI for Testcontainers, you can use `DockerClientFactory.instance().client()`. It will either resolve or throw an exception

Reply:
Yes but it would involve adding the test containers dependency to this project just for detection. This is not necessarily a bad thing but if I can limit the scope for now...